src/mapgen_v6.cpp

```diff
@@ -194,10 +213,9 @@ s16 MapgenV6::find_stone_level(v2s16 p2d)
 	s16 y;
 
 	for (y = y_nodes_max; y >= y_nodes_min; y--) {
-		MapNode &n = vm->m_data[i];
-		content_t c = n.getContent();
-		if (c != CONTENT_IGNORE && (
-			c == c_stone || c == c_desert_stone))
+		content_t c = vm->m_data[i].getContent();
+		if (c != CONTENT_IGNORE &&
+				(c == c_stone || c == c_desert_stone || c == c_dirt_with_grass))
```

Comment:
Wait, doesn't checking for dirt_with_grass defeat the purpose of find_stone_level? Dirt isn't stone.